test/isolation.test.ts

```diff
@@ -0,0 +1,38 @@
+import { dirname, resolve } from 'path'
+import { Tinypool } from 'tinypool'
+import { fileURLToPath } from 'url'
+
+const __dirname = dirname(fileURLToPath(import.meta.url))
+
+test('workers can be isolated between tasks', async () => {
+  const pool = new Tinypool({
+    filename: resolve(__dirname, 'fixtures/sleep.js'),
+    minThreads: 1,
+    maxThreads: 1,
+    isolateWorkers: false,
+  })
+
+  function getThreadId() {
+    expect(pool.threads).toHaveLength(1)
+    return pool.threads[0]!.threadId
+  }
+
+  const initialThreadId = getThreadId()
+
+  await pool.run({})
+  expect(getThreadId()).toBe(initialThreadId)
+
+  await pool.run({})
+  await pool.run({})
+  await pool.run({})
+  expect(getThreadId()).toBe(initialThreadId)
+
+  // TODO: pool.flushWorkers()
```

Comment:
@AriPerkkio  Have you checked my latest comment in the issue?

Reply:
The `filter` API is also a good idea. I answered more to the issue.